### ISSUE TITLE
Remove google_project.auto_create_network

### ISSUE THREAD

**Issue by paddycarver** [terraform-providers/terraform-provider-google#4348]
Now that we have `constraints/compute.skipDefaultNetworkCreation` as an option, can we get rid of the `auto_create_network` field and logic on `google_project`? This would offer a more approved way to skip the default network creation.

**Comment by Chupaka** [comment]
Isn't this Organization Policy? What about standalone projects, not tied to any Org?

**Comment by paddycarver** [comment]
Hey y'all, sorry for the lack of clarity here. Yes, this is talking about organization policy.

For context, I created this issue not because we have decided to do this, but because I want to remember to make an explicit decision when we're deciding what is and is not going into 3.0.0. Use cases this would cut off, like you shared, are very helpful in making that decision.

Right now I'd say probably don't worry about this, cutting off use cases with little gain except simplification is a tough sell, but I do want to talk about it with the other maintainers. I'd say going ahead with this is unlikely, but possible if we talk about it and find it's warranted or necessary.

Mostly, I just want to make an explicit decision and record it, instead of deciding by default by forgetting to decide anything. :)

**Comment by paddycarver** [comment]
So we discussed this change, and we decided that because not everyone has an org, and not everyone wants to manage this at the org-level (some users, reasonably, may want different per-project settings), we're going to keep this as-is. However, our ideal solution is for the API to create a project to support this functionality. I'm going to keep this open to track the work item of filing an internal issue to add that functionality to the API, which I think @danawillow is going to take on. When the issue has been filed, this can be closed, as we're not going to be doing it.
